src/main/java/org/prebid/server/bidder/rtbhouse/RtbhouseBidder.java

```diff
@@ -101,10 +106,60 @@ private static List<BidderBid> extractBids(BidResponse bidResponse) {
                 .filter(Objects::nonNull)
                 .flatMap(Collection::stream)
                 .filter(Objects::nonNull)
-                .map(bid -> BidderBid.of(bid, BidType.banner, bidResponse.getCur()))
+                .map(bid -> resolveBidderBid(bid, bidRequest, bidResponse, bidderErrors))
                 .toList();
     }
 
+    private BidderBid resolveBidderBid(Bid bid, BidRequest bidRequest,
+                BidResponse bidResponse, List<BidderError> bidderErrors) {
+        final String currency = bidResponse.getCur();
+        final BidType bidType = getBidType(bid.getImpid(), bidRequest.getImp());
+        final String bidAdm = bid.getAdm();
+        final String resolvedAdm = bidAdm != null && bidType == BidType.xNative
+                ? resolveNativeAdm(bidAdm, bidderErrors)
+                : bidAdm;
+        final Bid updatedBid = bid.toBuilder()
+                .adm(resolvedAdm)
+                .build();
+
+        return BidderBid.builder()
+                .bid(updatedBid)
+                .type(bidType)
+                .bidCurrency(currency)
+                .build();
+    }
+
+    private String resolveNativeAdm(String adm, List<BidderError> bidderErrors) {
+        final JsonNode admNode;
+        try {
+            admNode = mapper.mapper().readTree(adm);
+        } catch (JsonProcessingException e) {
+            bidderErrors.add(BidderError.badServerResponse("Unable to parse native adm: %s".formatted(adm)));
+            return null;
+        }
+
+        final JsonNode nativeNode = admNode.get("native");
+        if (!nativeNode.isMissingNode()) {
```

Comment:
Please, choose one of the solutions and use it here:
```
final JsonNode nativeNode = admNode.get("native");
if (nativeNode != null) {
```
or
```
private static final JsonPointer NATIVE_FIELD = JsonPointer.valueOf("/native");;
```
```
final JsonNode nativeNode = admNode.at(NATIVE_FIELD);
if (!nativeNode.isMissingNode()) {
```

Reply:
Fixed using the first solution.